packages/labs/context/src/test/late-provider_test.ts

```diff
@@ -300,4 +301,81 @@ suiteSkipIE('late context provider', () => {
     // Check that the consumer was called only once
     assert.equal(consumer.callCount, 1);
   });
+
+  test('a provider that upgrades after an ancestor provider', async () => {
+    const context = createContext<string>(Symbol());
+    @customElement('context-consumer-4')
+    class ContextConsumer4Element extends LitElement {
+      consume = new ContextConsumer(this, {
+        context,
+        subscribe: true,
+        callback: (value) => {
+          this.value = value;
+          this.callCount++;
+        },
+      });
+
+      value = 'consumer initializer';
+
+      callCount = 0;
+    }
+    @customElement('context-provider-grandparent')
+    class ContextProviderGrandparentElement extends LitElement {
+      provide = new ContextProvider(this, {
+        context,
+        initialValue: 'grandparent initial value',
+      });
+    }
+
+    container.innerHTML = `
+      <context-provider-grandparent>
+        <context-consumer-4></context-consumer-4>
+        <late-context-provider-4>
+          <context-consumer-4></context-consumer-4>
+        </late-context-provider-4>
+      </context-provider-grandparent>
+    `;
+    const directChildConsumer = container.querySelector(
+      'context-provider-grandparent > context-consumer-4'
+    ) as ContextConsumer4Element;
+    const indirectChildConsumer = container.querySelector(
+      'late-context-provider-4 > context-consumer-4'
+    ) as ContextConsumer4Element;
+    const grandparentProvider = container.querySelector(
+      'context-provider-grandparent'
+    ) as ContextProviderGrandparentElement;
+
+    await directChildConsumer.updateComplete;
+    assert.equal(directChildConsumer.value, 'grandparent initial value');
+    assert.equal(directChildConsumer.callCount, 1);
+    assert.equal(indirectChildConsumer.value, 'grandparent initial value');
+    assert.equal(indirectChildConsumer.callCount, 1);
+    grandparentProvider.provide.setValue('grandparent updated');
+    await directChildConsumer.updateComplete;
+    assert.equal(directChildConsumer.value, 'grandparent updated');
+    assert.equal(directChildConsumer.callCount, 2);
+    assert.equal(indirectChildConsumer.value, 'grandparent updated');
+    assert.equal(indirectChildConsumer.callCount, 2);
+
+    @customElement('late-context-provider-4')
+    class LateContextProvider4Element extends LitElement {
+      provide = new ContextProvider(this, {
+        context,
+        initialValue: 'late provider initial value',
+      });
+    }
+    await directChildConsumer.updateComplete;
+    // bad!
```

Comment:
I'm confused, why is this bad? (Same as the line beneath). These assertions seem to be the correct behavior that I would expect. The `indirectChildConsumer` starts listening to the `late-context-provider`, and the `directChildConsumer` to the `grandparentProvider`. 

It might be nice to add one more test which is that `setValue` on the grandparent provider will only update the `directChildConsumer`.

Reply:
Nit: remove the `// bad!` comments.

Reply:
Ah, forgot to upload!